src/base-client.ts

```diff
@@ -72,6 +73,83 @@ export class BaseSlackAPIClient implements BaseSlackClient {
     return await this.createBaseResponse(response);
   }
 
+  async fileUploadV2(
+    args: FileUploadV2Args,
+  ) {
+    const { file_uploads } = args;
+    const uploadUrls = await Promise.all(
+      file_uploads.map((file) => this.getFileUploadUrl(file)),
+    );
+
+    await Promise.all(
+      uploadUrls.map((uploadUrl, index) =>
+        this.uploadFile(uploadUrl.upload_url, file_uploads[index].file)
+      ),
+    );
+
+    return await Promise.all(
+      uploadUrls.map((uploadUrl, index) =>
+        this.completeFileUpload(uploadUrl.file_id, file_uploads[index])
+      ),
+    );
+  }
+
+  private async getFileUploadUrl(file: FileUploadV2) {
+    const fileMetaData = {
+      filename: file.filename,
+      length: file.length,
+      alt_text: file.alt_text,
+      snippet_type: file.snippet_type,
+    };
+    const response = await this.apiCall(
+      "files.getUploadURLExternal",
+      fileMetaData,
+    );
+
+    if (!response.ok) {
+      throw new Error(JSON.stringify(response.response_metadata));
+    }
+    return response;
+  }
+
+  private async completeFileUpload(fileID: string, file: FileUploadV2) {
+    const fileMetaData = {
+      files: JSON.stringify([{ id: fileID, title: file.title }]),
+      channel_id: file.channel_id,
+      initial_comment: file.initial_comment,
+      thread_ts: file.thread_ts,
+    };
+    const response = await this.apiCall(
+      "files.completeUploadExternal",
+      fileMetaData,
+    );
+    if (!response.ok) {
+      throw new Error(JSON.stringify(response.response_metadata));
+    }
+    return response;
+  }
+
+  private async uploadFile(
+    uploadUrl: string,
+    file: FileUploadV2["file"],
+  ) {
+    const response = await fetch(uploadUrl, {
+      headers: {
```

Comment:
When actually POSTing the file contents to the URL provided by `getUploadURLExternal`, we post the data as binary (raw bytes), or multipart-form-encoded (see the docs on this [here](https://api.slack.com/methods/files.getUploadURLExternal#markdown)).

As a result, I don't think we should be providing headers like Content-Type. The file-upload-external edge URL that we POST file data to will infer the content type based on the data uploaded. 

In [the node SDK, the headers are empty other than the token](https://github.com/slackapi/node-slack-sdk/blob/main/packages/web-api/src/WebClient.ts#L481C52-L486).
Similarly, [in the python SDK, same thing](https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/web/internal_utils.py#L381).

I think the trick in implementing this API is to ensure we encode the POST data in the appropriate format, based on what is provided by the user. Assuming we support the `content` parameter, like the other SDKs do, then the `content` parameter, which is a string, should be encoded as UTF-8 bytes. Otherwise, if `file` is a string, we should interpret that string as a file path, and read the raw bytes directly from that path. Finally, `file` may also be a buffer-like object. In this situation, we should provide the raw buffer directly to `fetch` as its data payload.

Reply:
OK. Indeed, delegating the setting of Content-Type to the fetch function is appropriate. Thank you for the detailed explanation 😊. Based on this, I will make the following modifications to the source code:

- When uploading a file to the URL obtained from getUploadURLExternal, do not set the Content-Type.
- Implement a content field, read its value as UTF-8, and pass it to the request body of the fetch function.
- When a string is passed to the file field, interpret it as a file path and read the file using Deno.readFileSync.